server.js

```diff
@@ -0,0 +1,132 @@
+const path = require('path')
+const fs = require('fs').promises
+
+// Import/export paths
+const srcIcons = path.join(__dirname, '/src/Icon')
+const destIcons = path.join(__dirname, '/build/lib')
+
+// File format
+const ext = '.svg'
+
+// SVGO settings
+const SVGO = require('svgo')
+const svgo = new SVGO({
+  multipass: true,
+  plugins: [
+    {
+      convertPathData: {
+        floatPrecision: 2,
+        transformPrecision: 4
+      }
+    },
+    {
+      cleanupNumericValues: {
+        floatPrecision: 2
+      }
+    },
+    {
+      collapseGroups: true
+    },
+    {
+      removeTitle: true,
+    },
+    {
+      removeViewBox: false,
+    },
+    {
+      removeUselessStrokeAndFill: true,
+    },
+    {
+      removeAttrs: {
+        attrs: ['xmlns', 'fill-rule', 'clip-rule']
+      }
+    },
+    {
+      mergePaths: {
+        force: true
+      }
+    }
+  ]
+});
+
+(async () => {
+  // Read the source directory of SVGs
+  let icons = await fs.readdir(srcIcons)
+
+  // We only want .svg, ignore the rest
+  icons = icons.filter(i => path.extname(i).toLowerCase() === ext)
+
+  // Remove .svg from name
+  icons = icons.map(i => path.basename(i, ext))
+
+  // Sort alphabetically
+  icons = icons.sort()
+
+  // Array of promises which do the fetching of the files
+  let processed = icons.map(i => fs.readFile(path.resolve(srcIcons, i + ext), 'utf8'))
+  processed = await Promise.all(processed)
+
+  // Optimise them with SVGO
+  processed = processed.map(i => svgo.optimize(i))
+  processed = await Promise.all(processed)
+
+  // Get the data from the SVGO object
+  processed = processed.map(i => i.data)
+
+  // Do our custom tweaks to the output
+  processed = processed.map((i, idx) =>
+    i
+      .replace('<svg', `<svg aria-hidden="true" class="svg-icon icon${icons[idx]}"`)
+      .replace(/<\/?g(\s.+?)*>/g, '')
+      .replace(/fill="#000"/gmi, '')
+      .replace(/fill="none"/gmi, '')
+      .replace(/fill="#222426"/gmi, 'fill="var(--black-800)"')
+      .replace(/fill="#fff"/gmi, 'fill="var(--white)"')
+      .replace(/fill="#6A7E7C"/gmi, 'fill="var(--black-500)"')
+      .replace(/fill="#1A1104"/gmi, 'fill="var(--black-900)"')
+      .replace(/\s>/gm, '>')
+      .replace(/\s\/>/gm, '/>')
+  )
+
+  // Make an object of our icons { IconName: '<svg>' }
+  let iconsObj = {}
+  processed.forEach((icon, idx) => {
+    iconsObj[icons[idx]] = icon
+
+    // Save each svg
+    fs.writeFile(path.resolve(destIcons, icons[idx] + ext), icon, 'utf8')
+  })
+
+  // Read the existing helper.js between "// Start icons" and "// End icons"
+  const jsFile = path.join(__dirname, '/build/helper.js')
```

Comment:
In a followup PR I'd like to see this being built and exported "properly". See my comment on setting the `main` value in package.json.

No action currently needed.

Reply:
I'd like to see this done in this PR if possible. ¯\_(ツ)_/¯

Reply:
I can do this in a followup PR. Ideally it'd bundle the js properly and export as a `umd` script. That way it can be either included in a browser (codepen) or imported in another file (webpack, node, etc). There's multiple ways to do this, but I don't know what the *lightest* way to do so is. My go-to is webpack, but Typescript would do it as well

Reply:
Added UMD support, it will load the helper for browsers or an object of the icons for anything else (CommonJS or AMD) - example in the readme!

Reply:
This is *much* closer to where I'd like to be, but at the end of the day we're still doing a string replace into a hand written file. Do you mind if I commit right to this PR? I can pop it out pretty quickly

Reply:
@b-kelly I think the Stacks "no one owns a PR" ethos can apply here. Commit away, let's get this awesomeness merged.

Reply:
Alright, I popped in "proper" bundling with Webpack. Was easier to pull that in and call than explore migrating to typescript, etc. Theoretically, this *should* be tree shakeable, but I haven't tested it yet. When in the browser, all items are bound to `window.StacksIcons` and the `browserHelper` will auto-execute (I *did* test this).

I put "proper" in quotes because David's solution was definitely acceptable, but was really only a small step up from the naive concat we were doing. The largest benefit here is that we can now import the json file directly if we wanted to, potentially allowing us to tree shake and drop any icons from inclusion that are not in use. After this gets merged, I can immediately test in one of my dependent projects and iterate if it doesn't work as planned.

I know it isn't super clean, so feel free to drop the commit if you like and I can go back to the drawing board.

Reply:
Love this, much more maintainable and extendible so think it's worth the extra dependancy 👍

Reply:
Just tried to test this locally - works great in the browser but with Node.JS I get:

> ReferenceError: window is not defined

Reply:
Fixed! Tested with nodejs, script tag and `import`ing into another project (thank you `npm link`!). All three work as expected, with `browserHelper` *only* running when added via the script tag. It doesn't seem that tree shaking works by default when importing, but I can worry about that another day.